lib/reply.js

```diff
@@ -216,35 +211,32 @@ Reply.prototype.removeHeader = function (key) {
   return this
 }
 
-Reply.prototype.header = function (key, value) {
-  const _key = key.toLowerCase()
-
-  // default the value to ''
-  value = value === undefined ? '' : value
+Reply.prototype.header = function (key, value = '') {
+  key = key.toLowerCase()
 
-  if (this[kReplyHeaders][_key] && _key === 'set-cookie') {
+  if (this[kReplyHeaders][key] && key === 'set-cookie') {
     // https://tools.ietf.org/html/rfc7230#section-3.2.2
-    if (typeof this[kReplyHeaders][_key] === 'string') {
-      this[kReplyHeaders][_key] = [this[kReplyHeaders][_key]]
+    if (typeof this[kReplyHeaders][key] === 'string') {
+      this[kReplyHeaders][key] = [this[kReplyHeaders][key]]
     }
+
     if (Array.isArray(value)) {
-      Array.prototype.push.apply(this[kReplyHeaders][_key], value)
+      this[kReplyHeaders][key].push(...value)
     } else {
-      this[kReplyHeaders][_key].push(value)
+      this[kReplyHeaders][key].push(value)
     }
   } else {
-    this[kReplyHeaders][_key] = value
+    this[kReplyHeaders][key] = value
   }
+
   return this
 }
 
 Reply.prototype.headers = function (headers) {
-  const keys = Object.keys(headers)
-  /* eslint-disable no-var */
-  for (var i = 0; i !== keys.length; ++i) {
-    const key = keys[i]
-    this.header(key, headers[key])
```

Comment:
`for in` is much faster than `Object.keys/entries()`

Reply:
Can you link to some evidence?

Reply:
[In browser tests](https://perf.link/#eyJpZCI6ImsxbmJoNDlxOXFxIiwidGl0bGUiOiJUZXN0IGZvciBsb29wcyIsImJlZm9yZSI6ImZ1bmN0aW9uIGdldFJhbmRvbVN0cigpIHtcbiAgICByZXR1cm4gTWF0aC5yYW5kb20oKS50b1N0cmluZygxNikuc3Vic3RyaW5nKDIpXG59XG5cbmNvbnN0IG1hcCA9IHt9O1xuXG5mb3IgKGxldCBpID0gMDsgaSA8IDEwOyBpKyspIHtcbiAgICBtYXBbZ2V0UmFuZG9tU3RyKCldID0gZ2V0UmFuZG9tU3RyKCk7XG59IiwidGVzdHMiOlt7Im5hbWUiOiJmb3IgaW4iLCJjb2RlIjoibGV0IHJlc3VsdCA9ICcnXG5cbmZvciAoY29uc3Qga2V5IGluIG1hcCkge1xuICAgIHJlc3VsdCArPSBrZXkgKyBtYXBba2V5XTtcbn0iLCJydW5zIjpbNjkwMDAsMTA1MDAwLDczODAwMCwxMTQ0MDAwLDMzMzAwMCwxMDAwLDg5MDAwLDY3NzAwMCw3OTgwMDAsOTQwMDAwLDU3MzAwMCw2OTUwMDAsNDQ0MDAwLDUwMDAsMTA1MTAwMCw1NjEwMDAsNjEwMDAsODAxMDAwLDI4OTAwMCwxMzEwMDAsMTAxMDAwMCw4NTkwMDAsNDI5MDAwLDEzMTAwMCw2NDYwMDAsNDAyMDAwLDEwOTEwMDAsOTE2MDAwLDc4ODAwMCw5NDcwMDAsMTM2MDAwLDE1MTAwMCw1MjQwMDAsODQzMDAwLDMzMzAwMCwxNzkwMDAsNDQyMDAwLDk3MDAwLDM3MzAwMCwzODAwMCw4MTMwMDAsNDU5MDAwLDU2MzAwMCwxMTcwMDAsNjA3MDAwLDI5MjAwMCwzNjIwMDAsMTYwMDAwLDIyODAwMCw0ODMwMDAsNTMwMDAsNzYzMDAwLDIyODAwMCwxMjAwMCwxMDAwMCw4NTkwMDAsMTA1MDAwMCwxNDEwMDAsOTM0MDAwLDg1OTAwMCw3OTgwMDAsNjA0MDAwLDcxNTAwMCw3MDAwLDg1OTAwMCw1MTIwMDAsMTY0MDAwLDY0ODAwMCw3MDAwMDAsNDU1MDAwLDg1OTAwMCw4NTkwMDAsMTMwMDAwLDg1OTAwMCw1MjYwMDAsMTE0NDAwMCw0MzEwMDAsMTE0NDAwMCw1MDQwMDAsMTA4ODAwMCwyMTQwMDAsMTM1MDAwLDEyMzAwMCwxOTAwMDAsNjM2MDAwLDg4MjAwMCw3NzgwMDAsODE5MDAwLDEyMTAwMCw3NDAwMCwyNDYwMDAsMjQ2MDAwLDMwMDAwLDYxMzAwMCw2NTEwMDAsMTA3NTAwMCw4NTkwMDAsOTc2MDAwLDU1ODAwMCw0NDQwMDBdLCJvcHMiOjUxNTA5MH0seyJuYW1lIjoiZm9yIC8gT2JqZWN0LmtleXMoKSIsImNvZGUiOiJsZXQgcmVzdWx0ID0gJydcbmNvbnN0IGtleXMgPSBPYmplY3Qua2V5cyhtYXApO1xuXG5mb3IgKGxldCBpID0gMDsgaSA8IGtleXMubGVuZ3RoOyBpKyspIHtcbiAgICByZXN1bHQgKz0ga2V5c1tpXSArIG1hcFtrZXlzW2ldXTtcbn0iLCJydW5zIjpbNTYxMDAwLDYyMzAwMCw0OTkwMDAsNjc3MDAwLDU1MDAwLDUwMDAwLDUxNjAwMCw3MTAwMDAsNjAyMDAwLDY0OTAwMCwyMjAwMDAsMTkzMDAwLDI1NjAwMCwzNDcwMDAsNDMwMDAsMzY0MDAwLDUxNjAwMCw3NDcwMDAsNjg4MDAwLDYyMzAwMCwyNTYwMDAsMjU5MDAwLDUwOTAwMCwzNDgwMDAsMTM4MDAwLDc0NzAwMCwyMzQwMDAsNjcwMDAsNjI0MDAwLDE2NjAwMCw3NDcwMDAsNDk4MDAwLDIyNTAwMCwxNTcwMDAsMTM2MDAwLDUwMDAwMCw1NjUwMDAsNTUwMDAsMzgwMDAwLDIxNTAwMCw1OTYwMDAsMjI5MDAwLDYyNDAwMCw0NTkwMDAsMjYwMDAwLDcwNDAwMCwzNDkwMDAsNTk0MDAwLDI1NDAwMCwxMDAwLDM1NTAwMCwzNjAwMDAsNjQ5MDAwLDMyMTAwMCwzMjgwMDAsNDA4MDAwLDQ1MTAwMCw0MDAwLDIwNTAwMCw2NDUwMDAsNjg2MDAwLDQ5MDAwMCw1NzIwMDAsMTY2MDAwLDEzMDAwMCwyNjkwMDAsMzY0MDAwLDI0ODAwMCwyNTcwMDAsMTU4MDAwLDE3MzAwMCwxMjkwMDAsMTEwMDAwLDQxMTAwMCwzOTQwMDAsNTM1MDAwLDM1NTAwMCw0OTAwMCwxMjkwMDAsMzA0MDAwLDYxODAwMCw0MTAwMCwzODAwMDAsNTc4MDAwLDI5NDAwMCwyNTkwMDAsMjM4MDAwLDIxNzAwMCw1OTAwMDAsNDI5MDAwLDQwMzAwMCwyMDAwLDEzMTAwMCwyNTMwMDAsNjcwMDAsNjM5MDAwLDY1NDAwMCw2NzAwMDAsNjU5MDAwLDU4MDAwXSwib3BzIjozNjY3MDB9LHsibmFtZSI6ImZvciBvZiAvIE9iamVjdC5lbnRyaWVzKCkiLCJjb2RlIjoibGV0IHJlc3VsdCA9ICcnXG5cbmZvciAoY29uc3QgW2tleSwgdmFsdWVdIG9mIE9iamVjdC5lbnRyaWVzKG1hcCkpIHtcbiAgICByZXN1bHQgKz0ga2V5ICsgdmFsdWU7XG59IiwicnVucyI6Wzc4MDAwLDExMDAwLDExMzAwMCwyNzAwMCwxMTUwMDAsNzgwMDAsMzEwMDAsNDMwMDAsMTQwMDAsMjA2MDAwLDYxMDAwLDc4MDAwLDE1NDAwMCwxMDAwLDcxMDAwLDIyNTAwMCw1MTAwMCwxMzcwMDAsMzMwMDAsMjQ5MDAwLDEzODAwMCw3ODAwMCwxNDAwMDAsNjIwMDAsMTE0MDAwLDE5MTAwMCw2ODAwMCwxODIwMDAsMjQ2MDAwLDQxMDAwLDI2NzAwMCwxMzYwMDAsMTEzMDAwLDYyMDAwLDQxMDAwLDI5NTAwMCwxOTAwMDAsMjA0MDAwLDU0MDAwLDIwMTAwMCw0NDAwMCw3ODAwMCwxNzUwMDAsMTc5MDAwLDUzMDAwLDYzMDAwLDE2OTAwMCw0MTAwMCwyNDcwMDAsODcwMDAsMTA4MDAwLDIyMTAwMCwxOTEwMDAsMjE4MDAwLDI0NjAwMCwxMTAwMDAsOTAwMCwxNjcwMDAsMTEzMDAwLDE5NjAwMCwyMjAwMCwxMzAwMDAsMzAwMDAsNjAwMCwxNTQwMDAsNzgwMDAsMTMzMDAwLDU4MDAwLDc4MDAwLDc0MDAwLDE5NzAwMCwyODcwMDAsNjIwMDAsNzgwMDAsNDMwMDAsMjIxMDAwLDc4MDAwLDIzMzAwMCwxMTkwMDAsMjI3MDAwLDIwNTAwMCw3MTAwMCwxMDcwMDAsMTM4MDAwLDE5MDAwLDY4MDAwLDE2NzAwMCw3ODAwMCw0NjAwMCwzMzAwMCw2MDAwLDc5MDAwLDEyNDAwMCwxMjMwMDAsMTY2MDAwLDUyMDAwLDIxMzAwMCwxNDIwMDAsNTAwMDAsNjQwMDBdLCJvcHMiOjExNTczMH0seyJuYW1lIjoiZm9yRWFjaCAvIE9iamVjdC5lbnRyaWVzKCkiLCJjb2RlIjoibGV0IHJlc3VsdCA9ICcnXG5cbk9iamVjdC5lbnRyaWVzKG1hcCkuZm9yRWFjaCgoW2tleSwgdmFsdWVdKSA9PiB7XG4gICAgcmVzdWx0ICs9IGtleSArIHZhbHVlO1xufSkiLCJydW5zIjpbMjAzMDAwLDk5MDAwLDE5OTAwMCwyNjcwMDAsNjkwMDAsMTMwMDAsODAwMDAsMjAwMCwyMTQwMDAsMTA2MDAwLDIzMjAwMCwxOTIwMDAsMTkwMDAsMTk5MDAwLDc1MDAwLDcxMDAwLDEyMTAwMCwxMTgwMDAsMzAwMDAsMjAwMDAwLDIxMDAwLDYzMDAwLDQwMDAsMTU1MDAwLDEzOTAwMCw5NDAwMCwyMDAwMDAsNjEwMDAsMjAwMDAwLDc4MDAwLDI1NDAwMCw4MTAwMCw1NjAwMCwyMDAwMDAsMjAwMDAwLDkzMDAwLDM2MDAwLDE2MDAwLDEyNjAwMCwxMzAwMCw2NjAwMCwxMDcwMDAsMTAwMCwxMjQwMDAsMTQ5MDAwLDE4MzAwMCwyMDAwMDAsNjEwMDAsMTQ4MDAwLDIwNjAwMCwxNjkwMDAsMTk1MDAwLDM1MDAwLDEyOTAwMCwyMDUwMDAsMTQ2MDAwLDEwMDAsMTc1MDAwLDE1MDAwLDI1MjAwMCwxMDAwLDEwMDAwLDI1MTAwMCwyMDAwLDg2MDAwLDE1MzAwMCwyMTcwMDAsMTE4MDAwLDE1OTAwMCwxMzkwMDAsMjAwMDAwLDI1MDAwLDMwMDAsNDkwMDAsNTMwMDAsOTAwMDAsMTQxMDAwLDE2NzAwMCwxOTgwMDAsMjY2MDAwLDEwODAwMCwyMDAwMDAsMjAwMCwzNTAwMCwyMDAwMDAsMjUwMDAwLDEwNTAwMCw1OTAwMCw4NjAwMCwxMjAwMDAsMTAwMCw3NzAwMCw1MzAwMCw5MjAwMCwxMzAwMCwxODYwMDAsNjkwMDAsMTAyMDAwLDEwMDAsOTAwMF0sIm9wcyI6MTEyNjIwfV0sInVwZGF0ZWQiOiIyMDIyLTA2LTIyVDEyOjQ2OjQ5LjczNloifQ%3D%3D) (node has similar results), **for in** is almost 2x faster then **for / Object.keys()** if there are not much elements in object (5 - 20) otherwise they have the same performance, so it make sense to leave more readable code, wdyt? `Object.entries()` is very slow as expected.

Reply:
We are not running in a browser.

Reply:
I also checked it in node, results are similar

Reply:
Can you post the benchmark across `node@14`, `node@16` and `node@18`. 
Because they use different version of `v8`. The optimization may not be the same.

Reply:
You can add the benchmark in: https://github.com/RafaelGSS/nodejs-bench-operations. It will run in a dedicated machine

Reply:
@climba03003 code for the benchmarks is here https://gist.github.com/denshakhov/469912ac59d07a07a4a5bf08114ace7a
![for-loops_node-v14 19 3](https://user-images.githubusercontent.com/107560822/175270325-84a5be93-ee22-4202-ba9d-c97bb6967944.png)
![for-loops_node-v16 15 1](https://user-images.githubusercontent.com/107560822/175270343-461b8a23-3e82-4893-972e-9ae75e5c05e0.png)
![for-loops_node-v18 4 0](https://user-images.githubusercontent.com/107560822/175270370-3eb74cb6-a1dd-4fe8-9581-c466c64bdbe7.png)